vizro-core/src/vizro/models/_controls/filter.py

```diff
@@ -15,19 +15,32 @@
 from vizro.managers import data_manager, model_manager
 from vizro.managers._model_manager import ModelID
 from vizro.models import Action, VizroBaseModel
-from vizro.models._components.form import Checklist, Dropdown, RadioItems, RangeSlider, Slider
+from vizro.models._components.form import (
+    Checklist,
+    DatePicker,
+    Dropdown,
+    RadioItems,
+    RangeSlider,
+    Slider,
+)
 from vizro.models._models_utils import _log_call
 from vizro.models.types import MultiValueType, SelectorType
 
-# TODO: Add temporal when relevant component is available
-SELECTOR_DEFAULTS = {"numerical": RangeSlider, "categorical": Dropdown}
+SELECTOR_DEFAULTS = {"numerical": RangeSlider, "categorical": Dropdown, "temporal": DatePicker}
 
 # Ideally we might define these as NumericalSelectorType = Union[RangeSlider, Slider] etc., but that will not work
 # with isinstance checks.
-SELECTORS = {"numerical": (RangeSlider, Slider), "categorical": (Checklist, Dropdown, RadioItems)}
+SELECTORS = {
+    "numerical": (RangeSlider, Slider),
+    "categorical": (Checklist, Dropdown, RadioItems),
+    "temporal": DatePicker,
+}
 
 
 def _filter_between(series: pd.Series, value: List[float]) -> pd.Series:
+    if len(value) != 2:
+        return pd.Series(False, index=series.index)
+
```

Comment:
Can you remind me - what does this case catch? It's late and my brain is already a bit foggy 😅

Reply:
My understanding is that if the date picker is deleted, the input `value` argument becomes `[None]`.

We should probably improve this validation and check if the value is a list with two elements where both elements are numbers.

Reply:
We should probably adjust value type hint accordingly then.

Reply:
Validations are no longer needed as date picker is no longer clearable.